frontend/packages/dev-console/integration-tests/plugins/index.js

```diff
@@ -2,26 +2,29 @@
 /* eslint-disable import/no-dynamic-require */
 const fs = require('fs');
 const webpack = require('@cypress/webpack-preprocessor');
+const NodePolyfillPlugin = require('node-polyfill-webpack-plugin');
```

Comment:
Will we be able to remove this when we upgrade node? We should make sure that it's tracked under some JIRA. Not sure offhand if we have a JIRA for node updates yet.

Reply:
likely not, this polyfills node APIs (path, process, etc.) used by cypress and the yaml language server

Reply:
OK, would be good to open a tech debt story to track the polyfill removal. Maybe when we get the rest of the dependencies up-to-date.

Reply:
see https://github.com/cypress-io/cypress/issues/8900#issuecomment-815032431 for cypress -- for the YAML language server, I think switching to https://github.com/remcohaszing/monaco-yaml (which also uses `yaml-language-server`) would remove the dependency on node APIs, but that is out of scope for this ticket

Reply:
> OK, would be good to open a tech debt story to track the polyfill removal. Maybe when we get the rest of the dependencies up-to-date.

https://issues.redhat.com/browse/CONSOLE-4360

Reply:
@spadgett @logonoff 

webpack 5 should provide [NodeStuffPlugin](https://github.com/webpack/webpack/blob/main/lib/NodeStuffPlugin.js) to polyfill or mock Node.js related globals.

https://webpack.js.org/configuration/node/

Also see https://webpack.js.org/configuration/resolve/#resolvefallback

> Webpack 5 no longer polyfills Node.js core modules automatically which means if you use them in your code running in browsers or alike, you will have to install compatible modules from npm and include them yourself. Here is a list of polyfills webpack has used before webpack 5:
> ```
> module.exports = {
>   //...
>   resolve: {
>     fallback: {
>       assert: require.resolve('assert'),
>       buffer: require.resolve('buffer'),
>       console: require.resolve('console-browserify'),
>       constants: require.resolve('constants-browserify'),
>       crypto: require.resolve('crypto-browserify'),
>       domain: require.resolve('domain-browser'),
>       events: require.resolve('events'),
>       http: require.resolve('stream-http'),
>       https: require.resolve('https-browserify'),
>       os: require.resolve('os-browserify/browser'),
>       path: require.resolve('path-browserify'),
>       punycode: require.resolve('punycode'),
>       process: require.resolve('process/browser'),
>       querystring: require.resolve('querystring-es3'),
>       stream: require.resolve('stream-browserify'),
>       string_decoder: require.resolve('string_decoder'),
>       sys: require.resolve('util'),
>       timers: require.resolve('timers-browserify'),
>       tty: require.resolve('tty-browserify'),
>       url: require.resolve('url'),
>       util: require.resolve('util'),
>       vm: require.resolve('vm-browserify'),
>       zlib: require.resolve('browserify-zlib'),
>     },
>   },
> };
> ```

Reply:
Should I switch to the `NodeStuffPlugin` module then? Using `NodePolyfillPlugin` just makes `resolve.fallback` cleaner and a bit easier to handle

Reply:
Seems like `NodeStuffPlugin` corresponds to the [`node`](https://webpack.js.org/configuration/node/) option in webpack, which only polyfills some constants like `global`, `__filename`, and `__dirname`:

https://github.com/mojir/webpack/blob/a1815f807a80e8c398510b600f9f6f2e1f40d905/lib/WebpackOptionsApply.js#L241

Unfortunately what we need from `NodePolyfillPlugin` are some node.js stdlib APIs such as `path`, `vm`, and `process` which are not provided by `NodeStuffPlugin`; ultimately the usage of these APIs should be eventually removed as well, and this is tracked in by [CONSOLE-4360](https://issues.redhat.com/browse/CONSOLE-4360)

Reply:
Thanks for looking into this :+1:  I'm OK with using `NodePolyfillPlugin` then.